src/service/repo/signedMessageRepo.ts

```diff
@@ -175,9 +179,16 @@ async function createSignedChat(members: string[]) {
   }
 }
 
-async function getSignedChannelDetail(channelType: GetstreamChannelType, channelId: string) {
+async function getSignedChannelDetail(
+  channelType: GetstreamChannelType,
+  channelId: string,
+  timeStamp?: string
+) {
+  const url = timeStamp
+    ? `${baseUrl.getSignedChannelDetail(channelType, channelId)}?last_fetch_date=${timeStamp}`
+    : baseUrl.getSignedChannelDetail(channelType, channelId);
   try {
-    const response = await api.get(baseUrl.getSignedChannelDetail(channelType, channelId));
+    const response = await api.get(url);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here introduce a new optional parameter `timeStamp` to the `getSignedChannelDetail` function. This is used to conditionally construct the URL for fetching channel details. However, there's no error handling in case the `api.get(url)` fails due to network issues or server errors. It would be good to add error handling to this function to improve its robustness.

```diff
  async function getSignedChannelDetail(
    channelType: GetstreamChannelType,
    channelId: string,
    timeStamp?: string
  ) {
    const url = timeStamp
      ? `${baseUrl.getSignedChannelDetail(channelType, channelId)}?last_fetch_date=${timeStamp}`
      : baseUrl.getSignedChannelDetail(channelType, channelId);
    try {
      const response = await api.get(url);
      if (response.status === 200) {
        return Promise.resolve(response.data?.data);
      }
+   } catch (error) {
+     console.error(`Failed to fetch signed channel detail: ${error}`);
+     return Promise.reject(error);
    }
  }
```
This will ensure that any errors during the API call are caught and handled appropriately.


<!-- This is an auto-generated comment by OSS CodeRabbit -->